### PR TITLE
CD-539

### PR DESCRIPTION
**Pull request by PivaFiras**
- Add Azure - Containers script
- Add script description JSON

### CODE REVIEW

File: examples/http/azure_containers.js [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+        if (!bodyAsJSON.value) {
+            D.failure(D.errorType.GENERIC_ERROR)
+            d.reject("No Container Groups found in the response");
+            return;
+        }
+        let containerGroupInfoList = bodyAsJSON.value.map(extractContainerGroupInfo);
+        if (!containerGroupInfoList.length) {
+            console.info('There is no Container Group');
+        } else {
+            containerGroupInfoList = filterContainerGroupInfoListByResourceGroups(filterContainerGroupInfoListByContainerNames(filterContainerGroupHadContainers(containerGroupInfoList)));
```
Comment (mohamedkallel) [inline examples/http/azure_containers.js:274]: why we should iterate over and over and over 3 times, we can iterate with only one filter and make our 3 conditions

File: examples/http/azure_containers.js [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+    return D.crypto.hash(value, "md5", null, "hex");
+}
+
+/**
+ * Inserts a containerGroup record into the containerGroup table with the given containerGroup information.
+ * @param {Object} container - The containerGroup information object.
+ * @returns {Promise} A promise that resolves when the record has been successfully inserted into the table.
+ */
+function insertRecord(container) {
+    const d = D.q.defer();
```
Comment (mohamedkallel) [inline examples/http/azure_containers.js:352]: why we are using promise for a simple insert to a table?

File: examples/http/azure_containers.js [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+ */
+function retrieveContainerGroupsPerformanceMetrics(containerGroupInfoList) {
+    const performanceKeyGroups = [];
+    const maxGroupSize = 20;
+    for (let i = 0; i < performanceMetrics.length; i += maxGroupSize) {
+        performanceKeyGroups.push(performanceMetrics.slice(i, i + maxGroupSize).map(function (metric) {
+            return metric.key
+        }).join(','));
+    }
+    const promises = containerGroupInfoList.map(function (containerGroupInfo) {
```
Comment (mohamedkallel) [inline examples/http/azure_containers.js:463]: please for promises use D.q and do not use Promise

create a promies variable as empty array and iterate over containerGroupInfoList and then iterate over performanceKeyGroups and in the second loop create a defer and push d.promise to your promises table

File: examples/http/azure_containers.js [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+        }
+        groupInfo.containers.forEach(function (containerInfo) {
+            containerDetailsExtractors.map(function (containerExtractor) {
+                containerGroupInfo[containerExtractor.key] = containerExtractor.extract(containerInfo);
+            })
+        })
+        return containerGroupInfo
+    })
+    const promises = containersListInfo.map(insertRecord);
+    return D.q.all(promises);
```
Comment (mohamedkallel) [inline examples/http/azure_containers.js:506]: why using promises for simple insert record to the table

**Review by mohamedkallel — CHANGES_REQUESTED** [review]

**Review by mohamedkallel — APPROVED** [review]

**Review by mohamedkallel — APPROVED** [review]

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**